lib/pysquared/usb/usbfunctions.py

```diff
@@ -0,0 +1,332 @@
+
+    def set_sd_file_system(self) -> None:
+        """Creates a file on the SD card."""
+        self.make_file("/data/temperature.txt")
+
+    def make_file(self, file_name: str, binary: bool = False) -> str:
+        """Creates a new file in the specified directory.
+
+        Args:
+            file_name (str): The name of the file to create.
+            binary (bool): If True, creates the file in binary mode.
+
+        Returns:
+            str: The name of the created file.
+        """
+        try:
+            ff = ""
+            n = 0
+            _folder = file_name[
+                : file_name.rfind("/") + 1
+            ]  # Extracts the folder from the file name
+            _file = file_name[
+                file_name.rfind("/") + 1 : file_name.rfind(".")
+            ]  # Extracts the file prefix from the file name
+            _file_type = file_name[
+                file_name.rfind(".") + 1 :
+            ]  # Extracts the file type from the file name
+            print(
+                "Creating new file in directory: /sd{} with file prefix: {}".format(
+                    _folder, _file
+                )
+            )
+            try:
+                os.chdir(
+                    "/sd" + _folder
+                )  # Changes the current directory to the specified folder
+            except OSError:
+                print("Directory {} not found. Creating...".format(_folder))
+                try:
+                    os.mkdir("/sd" + _folder)  # Creates the folder if it doesn't exist
+                except Exception as e:
+                    print(
+                        "Error with creating new file: "
+                        + "".join(traceback.format_exception(e))
+                    )
+                    return None
+            for i in range(
+                0xFFFF
+            ):  # Finding an available, unique file name by appending a unique number
+                ff = "/sd{}{}{:05}.{}".format(
+                    _folder, _file, (n + i) % 0xFFFF, _file_type
+                )
+                try:
+                    if n is not None:
+                        os.stat(ff)  # Checks if the file exists
+                except Exception:
+                    n = (n + i) % 0xFFFF
+                    # print('file number is',n)
+                    break
+            print("creating file..." + str(ff))
+            if binary:
+                b = "ab"  # Sets mode to binary append
+            else:
+                b = "a"  # Sets mode to append
+            with open(ff, b) as f:
+                f.tell()  # Returns current file position
+            os.chdir("/")  # Changes back to the root directory
+            return ff
+        except Exception as e:
+            print("Error creating file: " + "".join(traceback.format_exception(e)))
+            return None
+
+    def readfile(self, path: str, type: str = "string") -> str:
```

Comment:
Expanding on `__setitem__` complaint in sonarqube. Looks like this fn can return more than just a string.